compiler/rustc_hir_analysis/src/outlives/mod.rs

```diff
@@ -46,6 +47,24 @@ fn inferred_outlives_of(tcx: TyCtxt<'_>, item_def_id: LocalDefId) -> &[(ty::Clau
                 &[]
             }
         }
+        DefKind::Impl { of_trait: false }
```

Comment:
I'd prefer a solution that doesn't involve changing the rules for where we add implicit outlives obligations. This code is really delicate, and I'm worried this could turn out to be unsound without us really knowing why.

Is it possible to instead modify the object safety violation *error reporting* code to instead prefer pointing at `Self` type rather than arguments?

Reply:
Also, could you explain *why* adding this outlives obligation actually solves the problem? It's not necessarily clear why this is fixing what it does, and I'm not yet convinced its worth the changes to the `inferred_outlives_of` function for such a special case.